### ISSUE TITLE
[Feature Request] OPNsense Transfer rates/amounts

### ISSUE THREAD

**Issue by warmbo** [benphelps/homepage#470]
### Description

Being able to use the API in OPNsense to pull my total download/upload amounts or the current speed of my connection to see on Homepage would be fantastic.

### Other

https://github.com/opnsense/

**Comment by aidzer0** [comment]
This would be amazing. Came here to request this too, if possible.

**Comment by Oupsman** [comment]
I've made this, but for now it is definitely NOT ready for a PR because the code is ugly. And the memory usage reporting isn't reliable.  

![DeepinScreenshot_select-area_20221224091050](https://user-images.githubusercontent.com/11260343/209427226-59622ac7-827c-49df-91f0-c8039662b322.png)

**Comment by Oupsman** [comment]
Code is a little bit better, if you can try it out, it's there: https://github.com/Oupsman/homepage/tree/opnsense-widget

Config is : 

    - OPNSense:
        icon: opnsense.png
        href: url
        description: OPNSense
        widget:
          type: opnsense
          url: url
          username: key
          password: secret
          
You have to generate a key for this to work, the login/password you use to access the web interface will not work.

**Comment by aidzer0** [comment]
> 

Good work, first the Omada stuff now this, very well done!
I do not mind testing if needed, but I am absolutely lost on how to compose your packages and test it, I am very new to GitHub and its workings. If needed, just let me know and I will try to figure it out.
Take care and happy holidays!

**Comment by Oupsman** [comment]
How are you running your actual Homepage ?

**Comment by aidzer0** [comment]
> How are you running your actual Homepage ?

Docker on unraid.

**Comment by Oupsman** [comment]
Well, if the project's maintainers are OK with it, I can build a testing docker container and making it available on docker hub. With a name like 'homepage-opnsense-test'

**Comment by shamoon** [comment]
Sure though once you open a PR we can test it and will merge into `main` which will create an image for other folks test easily before it’s release fully

**Comment by Oupsman** [comment]
Sure, I can make a PR and I would love to see your comments on the code :)

**Comment by aidzer0** [comment]
Fantastic work! I just added the widget, all seems to be working, but the tags show opnsense.CPU, etc: 
<img width="544" alt="Screen Shot 2022-12-26 at 10 47 53" src="https://user-images.githubusercontent.com/56331974/209569028-0fc56214-dca5-464e-9eac-8b42c449efce.png">

Oupsman you literally added 3 widgets that I was hoping to be added. Well done good sir.

**Comment by shamoon** [comment]
Recreating the container should fix the labels

**Comment by aidzer0** [comment]
> Recreating the container should fix the labels

Yes, yes it did, thank you!!!

**Comment by aidzer0** [comment]
> Recreating the container should fix the labels

Seems that it worked until I restarted the container to load the image/icon files. once I restart the container to load images and icons it reverts back to the opnsense.CPU etc tags. 

It seems that when I re-install the container, the setting.yml file gets ignored and the tags on both opnSense and omada widgets load fine, but as soon as I restart the container to load the settings.yml, it shows the wrong tags.

I have tried fully removing the container and re-installing with no success :(

I will continue to troubleshoot later today to see if I can find a solution.
